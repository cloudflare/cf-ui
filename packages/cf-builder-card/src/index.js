import cardActions from './cardActions';
import CardActionTypes from './CardActionTypes';
import CardBuilder from './CardBuilder';
import cardReducer from './cardReducer';

export { cardActions, CardActionTypes, CardBuilder, cardReducer };
