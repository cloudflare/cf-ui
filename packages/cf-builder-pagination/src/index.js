import PaginationBuilder from './PaginationBuilder';

export { PaginationBuilder };
