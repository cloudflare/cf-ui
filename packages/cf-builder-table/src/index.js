import tableActions from './tableActions';
import TableActionTypes from './TableActionTypes';
import TableBuilder from './TableBuilder';
import TableBuilderPropTypes from './TableBuilderPropTypes';
import tableReducer from './tableReducer';

export {
  tableActions,
  TableActionTypes,
  TableBuilder,
  TableBuilderPropTypes,
  tableReducer
};
