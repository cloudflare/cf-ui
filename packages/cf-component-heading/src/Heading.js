import React, { PropTypes } from 'react';
import { createComponent } from 'cf-style-container';

const styles = ({ theme, size }) => {
  const t = {};

  if (theme[`fontWeight${size}`]) {
    t.fontWeight = theme[`fontWeight${size}`];
  }

  if (theme[`fontSize${size}`]) {
    t.fontSize = theme[`fontSize${size}`];
  }

  if (theme[`lineHeight${size}`]) {
    t.lineHeight = theme[`lineHeight${size}`];
  }

  if (theme[`marginTop${size}`]) {
    t.marginTop = theme[`marginTop${size}`];
  }
  return t;
};

const Heading = ({ size, className, children }) => {
  const tagName = 'h' + size;
  return React.createElement(tagName, { className }, children);
};

Heading.propTypes = {
  size: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
  className: PropTypes.string.isRequired,
  children: PropTypes.node
};

export default createComponent(styles, Heading);
