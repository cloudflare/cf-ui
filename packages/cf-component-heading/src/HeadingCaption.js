import React, { PropTypes } from 'react';
import { createComponent } from 'cf-style-container';

const styles = props => {
  const theme = props.theme;
  return {
    color: theme.color,
    fontSize: theme.fontFize,
    fontColor: theme.fontColor,
    fontWeight: theme.fontWeight,
    marginLeft: theme.marginLeft
  };
};

const HeadingCaption = ({ className, children }) => (
  <small className={className}>
    {children}
  </small>
);

HeadingCaption.propTypes = {
  className: PropTypes.string.isRequired,
  children: PropTypes.node
};

export default createComponent(styles, HeadingCaption);
