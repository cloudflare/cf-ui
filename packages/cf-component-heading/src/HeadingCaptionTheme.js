import { checkBaseTheme } from 'cf-style-const';

export default baseTheme => {
  checkBaseTheme(baseTheme, 'HeadingCaptionTheme');
  return {
    color: baseTheme.colorDarkGray,
    fontSize: '1rem',
    fontColor: baseTheme.fontColorHeadingCaption,
    fontWeight: 'lighter',
    marginLeft: '.3rem'
  };
};
