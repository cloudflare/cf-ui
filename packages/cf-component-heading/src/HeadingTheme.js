export default () => ({
  fontSize1: '2rem',
  lineHeight1: 1.2,
  fontWeight1: 700,
  fontSize2: '1.6rem',
  lineHeight2: 1.3,
  fontWeight3: 700,
  fontSize3: '1.46667rem',
  lineHeight3: 1.3,
  fontWeight3: 700,
  fontSize4: '1.2rem',
  lineHeight4: 1.3,
  fontWeight4: 700,
  fontSize5: '1rem',
  lineHeight5: 1.5,
  fontWeight5: 700,
  fontSize6: '.93333rem',
  lineHeight6: 1.6,
  fontWeight6: 700
});
