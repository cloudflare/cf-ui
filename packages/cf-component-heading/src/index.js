import Heading from './Heading';
import HeadingCaption from './HeadingCaption';

export { Heading, HeadingCaption };
