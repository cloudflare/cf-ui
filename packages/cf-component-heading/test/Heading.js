import React from 'react';
import { Heading } from '../../cf-component-heading/src/index';
import felaSnapshot from 'cf-style-provider/src/felaSnapshot';

test('should render size', () => {
  const snapshot = felaSnapshot(<Heading size={2}>Heading</Heading>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
