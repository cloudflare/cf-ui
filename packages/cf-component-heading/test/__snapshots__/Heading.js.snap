// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render size 1`] = `
<h2
  className="cf-46exok"
>
  Heading
</h2>
`;
